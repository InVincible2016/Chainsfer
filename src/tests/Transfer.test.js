import until from 'enzyme-shallow-until'
import ShallowWrapper from 'enzyme/ShallowWrapper'

import Transfer from '../components/TransferComponent'
import WalletSelection from '../containers/WalletSelectionContainer'

ShallowWrapper.prototype.until = until

describe('TransferComponent', () => {
  const history = { location: '/transfer' }

  it('step = 0', () => {
    const wrapper = shallow(
      <Transfer step={0} history={history} />
    ).until(WalletSelection)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('step = 1', () => {
    const wrapper = shallow(
      <Transfer step={1} history={history} />
    ).until(WalletSelection)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('step = 2', () => {
    const wrapper = shallow(
      <Transfer step={2} history={history} />
    ).until(WalletSelection)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('step = 3', () => {
    const wrapper = shallow(
      <Transfer step={3} history={history} />
    ).until(WalletSelection)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('step = 4', () => {
    const wrapper = shallow(
      <Transfer step={4} history={history} />
    ).until(WalletSelection)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
