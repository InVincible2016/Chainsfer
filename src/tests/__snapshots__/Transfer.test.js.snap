// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransferComponent step = 0 1`] = `
<div
  className="MuiGrid-container-3 MuiGrid-direction-xs-column-6"
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(MyStepper)
      actionType="transfer"
      step={0}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(Grid)
        className="TransferComponent-subComponent-2"
        item={true}
      >
        <Connect(WalletSelectionContainer) />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`TransferComponent step = 1 1`] = `
<div
  className="MuiGrid-container-3 MuiGrid-direction-xs-column-6"
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(MyStepper)
      actionType="transfer"
      step={1}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(Grid)
        className="TransferComponent-subComponent-2"
        item={true}
      >
        <Connect(RecipientContainer) />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`TransferComponent step = 2 1`] = `
<div
  className="MuiGrid-container-3 MuiGrid-direction-xs-column-6"
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(MyStepper)
      actionType="transfer"
      step={2}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(Grid)
        className="TransferComponent-subComponent-2"
        item={true}
      >
        <Connect(ReviewContainer) />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`TransferComponent step = 3 1`] = `
<div
  className="MuiGrid-container-3 MuiGrid-direction-xs-column-6"
>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(MyStepper)
      actionType="transfer"
      step={3}
    />
  </WithStyles(Grid)>
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(Grid)
        className="TransferComponent-subComponent-2"
        item={true}
      >
        <Connect(ReceiptContainer) />
      </WithStyles(Grid)>
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;

exports[`TransferComponent step = 4 1`] = `
<div
  className="MuiGrid-container-3 MuiGrid-direction-xs-column-6 TransferComponent-rootReceipt-1"
>
  <WithStyles(Grid)
    item={true}
  />
  <WithStyles(Grid)
    item={true}
  >
    <WithStyles(Grid)
      alignItems="center"
      container={true}
      direction="column"
    >
      <WithStyles(Grid)
        className="TransferComponent-subComponent-2"
        item={true}
      />
    </WithStyles(Grid)>
  </WithStyles(Grid)>
</div>
`;
